static/app/components/draggableTabs/draggableTabList.tsx

```diff
@@ -223,14 +249,51 @@ function BaseTabList({
 
 const collectionFactory = (nodes: Iterable<Node<any>>) => new ListCollection(nodes);
 
+export interface DraggableTabListProps
+  extends AriaTabListOptions<DraggableTabListItemProps>,
+    TabListStateOptions<DraggableTabListItemProps>,
+    Omit<DroppableCollectionStateOptions, 'collection' | 'selectionManager'> {
+  setTabs: (tabs: Tab[]) => void;
+  tabs: Tab[];
+  className?: string;
+  hideBorder?: boolean;
+  outerWrapStyles?: React.CSSProperties;
+}
+
 /**
  * To be used as a direct child of the <Tabs /> component. See example usage
  * in tabs.stories.js
  */
-export function TabList({items, ...props}: TabListProps) {
-  /**
-   * Initial, unfiltered list of tab items.
-   */
+export function DraggableTabList({
+  items,
+  tabs,
+  setTabs,
+  ...props
+}: DraggableTabListProps) {
+  const onInsert = async (e: DroppableCollectionInsertDropEvent) => {
+    const dropItem = e.items[0] as TextDropItem;
+    const eventTab: {key: string; value: string} = JSON.parse(
+      await dropItem.getText('tab')
+    );
+    const draggedTab = tabs.find(tab => tab.key === eventTab.key);
+    if (draggedTab) {
+      const updatedTabs = tabs.filter(tab => tab.key !== draggedTab.key);
+      const targetTab = tabs.find(tab => tab.key === e.target.key);
+      if (targetTab) {
+        if (e.target.key === draggedTab.key) {
+          return; // Do nothing if the dragged tab is dropped on itself
+        }
+        const targetIdx = updatedTabs.indexOf(targetTab);
+        if (e.target.dropPosition === 'before') {
+          updatedTabs.splice(targetIdx, 0, draggedTab);
+        } else if (e.target.dropPosition === 'after') {
+          updatedTabs.splice(targetIdx + 1, 0, draggedTab);
+        }
+        setTabs(updatedTabs);
+      }
+    }
```

Comment:
i think we can simplify stuff here by returning early and combining the find and indexOf
```suggestion
    const draggedTab = tabs.find(tab => tab.key === eventTab.key);
    if (!draggedTab || e.target.key === draggedTab.key) {
      return; // Do nothing if the dragged tab is dropped on itself
    }

    const updatedTabs = tabs.filter(tab => tab.key !== draggedTab.key);
    const targetIdx = tabs.findIndex(tab => tab.key === e.target.key);
    if (targetIdx > -1) {
      if (e.target.dropPosition === 'before') {
        updatedTabs.splice(targetIdx, 0, draggedTab);
      } else if (e.target.dropPosition === 'after') {
        updatedTabs.splice(targetIdx + 1, 0, draggedTab);
      }
      setTabs(updatedTabs);
    }
```